R/impute.R

```diff
@@ -0,0 +1,32 @@
+#' Imputation for modelling using mdates
+#'
+#' @name impute
+#' @param .data A dataset with mdate variables.
+#' @param times Numebr of times for imputation.
+#' @param imputes Imputed data.
+#' @import dplyr
+#' @examples
+#' lapply(impute(battles, 3), function(x) mean(x$Date))
+#' #library(psfmi)
+#' #res <- to_pool(impute(battles, 10)) |>
+#' #   dplyr::mutate(Date = as.numeric(Date)) |>
+#' #   psfmi_coxr(Surv(Date, status) ~ US_party + N_actors, impvar = "impute_num")
+#' #res$RR_model_final
+NULL
+
+#' @rdname impute
+#' @export
+impute <- function(.data, times = 1000) {
+  lapply(seq.int(times), function(x){
+    .data |> dplyr::mutate(dplyr::across(dplyr::where(is_messydate), random))
```

Comment:
While this works correctly, since we may yet change the function name, arguments etc, perhaps we should remove it from this release too?